Ntuplizer/plugins/JpsiMuNtuplizer.cc

```diff
@@ -317,400 +318,478 @@ particle_cand JpsiMuNtuplizer::calculateIPvariables(
-
-  event.getByToken(HLTtriggersToken_, HLTtriggers_);
-
-  bool isTriggered = false;
-  const edm::TriggerNames& trigNames = event.triggerNames(*HLTtriggers_);
-  std::string finalTriggerName = "";
+    /********************************************************************
+     *
+     * Step1: check if the J/psi trigger is fired.
+     * Namely, HLT_DoubleMu4_JpsiTrk_Displaced_v
+     * and  HLT_Dimuon0_Jpsi3p5_Muon2_v
+     ********************************************************************/
+
+    event.getByToken(HLTtriggersToken_, HLTtriggers_);
+
+    bool isTriggered = false;
+    const edm::TriggerNames& trigNames = event.triggerNames(*HLTtriggers_);
+    std::string finalTriggerName="";
+    std::string finalTriggerFilterObjName="";
+
+    for (unsigned int i = 0, n = HLTtriggers_->size(); i < n; ++i) {
+
+        // if(trigNames.triggerName(i).find("HLT_DoubleMu4_JpsiTrk_Displaced_v")!= std::string::npos || trigNames.triggerName(i).find("HLT_Dimuon0_Jpsi3p5_Muon2_v")!= std::string::npos ){
+           
+        //     nBranches_->HLT_BPH_isFired[trigNames.triggerName(i)] = HLTtriggers_->accept(i);
+        if(trigNames.triggerName(i).find("HLT_DoubleMu4_JpsiTrk_Displaced_v")!= std::string::npos){
+            nBranches_->HLT_BPH_isFired[trigNames.triggerName(i)] = HLTtriggers_->accept(i);
+            if(HLTtriggers_->accept(i)){
+                isTriggered = true;
+                finalTriggerName=trigNames.triggerName(i);  
+                finalTriggerFilterObjName="hltJpsiTkVertexFilter";
+                // std::cout << "finalTriggerName = "  << finalTriggerName << std::endl;
+               
+            }
+        }
 
-  for (unsigned int i = 0, n = HLTtriggers_->size(); i < n; ++i) {
-
-    if(trigNames.triggerName(i).find("HLT_DoubleMu4_JpsiTrk_Displaced_v")!= std::string::npos){
-      if(HLTtriggers_->accept(i)){
-	isTriggered = true;
-	finalTriggerName = trigNames.triggerName(i);
-      }
     }
-  }
-
-  if(!isTriggered) return;
-
-  //  std::cout << "finalTriggerName = "  << finalTriggerName << std::endl;
-
-
-
-  /********************************************************************
-   *
-   * Step2: pre-select muons for building J/psi candidates ... 
-   * For muons, no requirement applied
-   *
-   ********************************************************************/
-
-  event.getByToken(verticeToken_   , vertices_     );
-  event.getByToken(bsToken_   , beamspot_     );
-  event.getByToken(muonToken_	, muons_    );
-  event.getByToken(triggerObjects_  , triggerObjects);
-
-  std::vector<pat::Muon> muoncollection;
-  std::vector<int> muoncollection_id;
-  muoncollection.clear();
-  muoncollection_id.clear();
-
-  for(size_t imuon = 0; imuon < muons_->size(); ++ imuon){
+    // Second trigger if fist one didn't fire
+    if (!isTriggered){
+        for (unsigned int i = 0, n = HLTtriggers_->size(); i < n; ++i) {
+            if(trigNames.triggerName(i).find("HLT_DoubleMu4_JpsiTrk_Displaced_v")!= std::string::npos){
```

Comment:
This seems to be duplicated to L342. You wanted to put another HLT path name here ?

Reply:
Hi Yuta, this is the second trigger that was suggested by Palla and Co. My idea was to save events that passed this second trigger, in case the other one is not fired.

Reply:
Yeah, but the name seems to be exactly the same to me ...: HLT_DoubleMu4_JpsiTrk_Displaced_v*